libweb3jsonrpc/Test.cpp

```diff
@@ -33,6 +34,64 @@ using namespace jsonrpc;
 
 Test::Test(eth::Client& _eth): m_eth(_eth) {}
 
+Json::Value fillJsonWithState(eth::State const& _state)
+{
+    Json::Value oState;
+    for (auto const& a : _state.addresses())
+    {
+        Json::Value o;
+        o["balance"] = toCompactHexPrefixed(_state.balance(a.first), 1);
+        o["nonce"] = toCompactHexPrefixed(_state.getNonce(a.first), 1);
+
+        Json::Value store;
+        for (auto const& s : _state.storage(a.first))
+            store[toCompactHexPrefixed(s.second.first, 1)] =
+                toCompactHexPrefixed(s.second.second, 1);
+        o["storage"] = store;
+
+        if (_state.code(a.first).size() > 0)
+            o["code"] = toHexPrefixed(_state.code(a.first));
+        else
+            o["code"] = "";
+
+        oState[toHexPrefixed(a.first)] = o;
+    }
+    return oState;
+}
+
+string exportLog(eth::LogEntries const& _logs)
+{
+    RLPStream s;
+    s.appendList(_logs.size());
+    for (eth::LogEntry const& l : _logs)
+        l.streamRLP(s);
+    return toHexPrefixed(sha3(s.out()));
+}
+
+const int c_postStateJustHashVersion = 1;
+const int c_postStateFullStateVersion = 2;
+const int c_postStateLogHashVersion = 3;
+string Test::test_getPostState(Json::Value const& param1)
+{
+    Json::Value out;
+    Json::FastWriter fastWriter;
+    if (u256(param1["version"].asString()) == c_postStateJustHashVersion)
```

Comment:
Does the RPC method take `1`, `2`, or `3`, indicating different versions?  In other parts of the RPC, "latest" means "latest", and so on.  I guess strings like "jushash" "fullstate" or "loghash" are easier.

Reply:
no problem here. it could be anything.
the discussion is here 
https://github.com/ethereum/retesteth/issues/7